File: batch/webCrawler/internal/model/guitar.go
package model

import "fmt"

type Guitar struct {
    Maker             int     `db:"maker"               json:"maker"`
    Name              string  `db:"name"                json:"name"`
    BodyFinish        string  `db:"body_finish"         json:"body_finish"`
    BodyMaterial      string  `db:"body_material"       json:"body_material"`
    BodyMaterialFront int     `db:"body_material_front" json:"body_material_front"`
    BodyMaterialBack  int     `db:"body_material_back"  json:"body_material_back"`
    Bridge            string  `db:"bridge"              json:"bridge"`
    Color             string  `db:"color"               json:"color"`
    Controls          string  `db:"controls"            json:"controls"`
    Comment           string  `db:"comment"             json:"comment"`
    Fingerboard       int     `db:"fingerboard"         json:"fingerboard"`
    FretCount         int     `db:"fret_count"          json:"fret_count"`
    Inlays            string  `db:"inlays"              json:"inlays"`
    Joint             string  `db:"joint"               json:"joint"`
    NeckMaterial      int     `db:"neck_material"       json:"neck_material"`
    Pickups           string  `db:"pickups"             json:"pickups"`
    Price             int     `db:"price"               json:"price"`
    ScaleLengthMM     int     `db:"scale_length_mm"     json:"scale_length_mm"`
    Series            string  `db:"series"              json:"series"`
    Src               string  `db:"src"                 json:"src"`
    Weight            float64 `db:"weight"              json:"weight"`
}


func (g *Guitar) String() string {
	return fmt.Sprintf("maker: %v, name: %v, price: %d", g.Maker, g.Name, g.Price)
}

File: frontend/src/types/Guitar.ts
export interface GuitarsRequest {
    makerCd?:            number;
    name?:               string;
    series?:             string;
    colorCd?:            number;
    bodyMaterialTopCd?:  number;
    bodyMaterialBackCd?: number;
    minPrice?:           number;
    maxPrice?:           number;
    order?:              string;
    sort?:               string;
    page?:               number;
    pageSize?:           number;
};

export interface GuitarsResponse {
    totalCount: number;
    page:       number;
    pageSize:   number;
    totalPages: number;
    hasPrev:    boolean;
    hasNext:    boolean;
    guitars:    Guitar[];
}

export interface Guitar {
    maker:            number;
    makerName:        string;
    name:             string;
    color:            string;
    colorCd:          number;
    bodyFinish:       string;
    bodyMaterial:     string;
    bodyMaterialTop:  number;
    bodyMaterialBack: number;
    bridge:           string;
    controls:         string;
    comment:          string;
    fingerboard:      number;
    fingerboardName:  string;
    fretCount:        number;
    inlays:           string;
    joint:            string;
    neckMaterial:     number;
    neckMaterialName: number;
    pickups:          string;
    price:            number;
    scaleLengthMm:    number;
    series:           string;
    src:              string;
    weight:           number;
    updated:          string;
};

export type GuitarParams = {
    makerCd:            number;
    name:               string;
    colorCd:            number;
    series:             string;
    bodyMaterialTopCd:  number;
    bodyMaterialBackCd: number;
    minPrice:           number;
    maxPrice:           number;
    order:              string;
    sort:               string;
    page:               number;
    pageSize:           number;

    setMakerCd:            React.Dispatch<React.SetStateAction<number>>;
    setName:               React.Dispatch<React.SetStateAction<string>>;
    setColorCd:            React.Dispatch<React.SetStateAction<number>>;
    setSeries:             React.Dispatch<React.SetStateAction<string>>
    setBodyMaterialTopCd:  React.Dispatch<React.SetStateAction<number>>;
    setBodyMaterialBackCd: React.Dispatch<React.SetStateAction<number>>;
    setMinPrice:           React.Dispatch<React.SetStateAction<number>>;
    setMaxPrice:           React.Dispatch<React.SetStateAction<number>>;
    setOrder:              React.Dispatch<React.SetStateAction<string>>;
    setSort:               React.Dispatch<React.SetStateAction<string>>;
    setPage:               React.Dispatch<React.SetStateAction<number>>;
    setPageSize:           React.Dispatch<React.SetStateAction<number>>;
};